hs_core/hydroshare/resource.py

```diff
@@ -747,6 +749,7 @@ def publish_resource(pk):
     resource.raccess.immutable = True
     resource.raccess.save()
     resource.doi = "to be assigned"
+    resource.metadata.create_element('date', type='published', start_date=now().isoformat())
```

Comment:
The resource_modified() function is called in many different situations (add file, delete file, update/create metadata elements). I am not sure why it would be more appropriate to create the 'published' date element in resource_modified() rather than in publish_resource().